wsmaster/che-core-api-factory/src/main/java/org/eclipse/che/api/factory/server/FactoryManager.java

```diff
@@ -133,8 +141,12 @@ public Factory updateFactory(Factory update, Set<FactoryImage> images) throws Co
                                                                                   ServerException {
         requireNonNull(update);
         final AuthorImpl creator = factoryDao.getById(update.getId()).getCreator();
+        FactoryImpl updateImpl = new FactoryImpl(update, images);
+        if (isNullOrEmpty(updateImpl.getName())) {
```

Comment:
For me it looks stupid, if you deleting the factory name on Dashb, and it returns back after update :). But seems others are with you, so ok.